src/lib.rs

```diff
@@ -378,47 +379,50 @@ pub fn verify_file_checksum(expected_checksum: &str, file_path: &str) -> Result<
     Ok(computed_checksum == expected_checksum)
 }
 
-/// Asynchronously downloads a file from a given URL to a specified destination path.
+/// Sets up the environment variables required for the ESP-IDF build system.
 ///
-/// # Arguments
-///
-/// * `url` - A string representing the URL from which to download the file.
-/// * `destination_path` - A string representing the path to which the file should be downloaded.
-/// * `show_progress` - A function pointer to a function that will be called to show the progress of the download.
-///
-/// # Returns
-///
-/// * `Ok(())` if the file was successfully downloaded.
-/// * `Err(std::io::Error)` if an error occurred during the download process.
-///
-/// # Example
+/// # Parameters
 ///
-/// ```rust
-/// use std::io;
-/// use std::io::Write;
-/// use idf_im_lib::download_file;
-///
-/// fn download_progress_callback(downloaded: u64, total: u64) {
-///     let percentage = (downloaded as f64 / total as f64) * 100.0;
-///     print!("\rDownloading... {:.2}%", percentage);
-///     io::stdout().flush().unwrap();
-/// }
+/// * `tool_install_directory`: A reference to a `PathBuf` representing the directory where the ESP-IDF tools are installed.
+/// * `idf_path`: A reference to a `PathBuf` representing the path to the ESP-IDF framework directory.
 ///
-/// #[tokio::main]
-/// async fn main() {
-///     let url = "https://example.com/file.zip";
-///     let destination_path = "/path/to/destination";
+/// # Return
 ///
-///     match download_file(url, destination_path, &download_progress_callback).await {
-///         Ok(()) => println!("\nDownload completed successfully"),
-///         Err(e) => eprintln!("Error during download: {}", e),
-///     }
-/// }
-/// ```
+/// * `Result<Vec<(String, String)>, String>`:
+///   - On success, returns a `Vec` of tuples containing the environment variable names and their corresponding values.
+///   - On error, returns a `String` describing the error.
+///
+pub fn setup_environment_variables(
+    tool_install_directory: &PathBuf,
+    idf_path: &PathBuf,
+) -> Result<Vec<(String, String)>, String> {
+    let mut env_vars = vec![];
+
+    // env::set_var("IDF_TOOLS_PATH", tool_install_directory);
+    let instal_dir_string = tool_install_directory.to_str().unwrap().to_string();
+    env_vars.push(("IDF_TOOLS_PATH".to_string(), instal_dir_string));
+    let idf_path_string = idf_path.to_str().unwrap().to_string();
+    env_vars.push(("IDF_PATH".to_string(), idf_path_string));
+
+    let python_env_path_string = tool_install_directory
+        .join("python")
+        .to_str()
+        .unwrap()
+        .to_string();
+    env_vars.push(("IDF_PYTHON_ENV_PATH".to_string(), python_env_path_string));
+
+    Ok(env_vars)
+}
+pub enum DownloadProgress {
+    Progress(u64, u64), // (downloaded, total)
+    Complete,
+    Error(String),
+}
+
 pub async fn download_file(
```

Comment:
this function was changed and before merge to the master the CLI part should be modified